stellar-contract-env-host/src/host.rs

```diff
@@ -177,6 +177,27 @@ struct RollbackPoint {
     objects: usize,
 }
 
+#[cfg(feature = "testutils")]
+pub trait ContractVTableFnTrait {
+    fn call(&self, args: &[RawVal]) -> RawVal;
+
+    fn duplicate(&self) -> ContractVTableFn;
+}
+
+#[cfg(feature = "testutils")]
+pub struct ContractVTableFn(pub Box<dyn ContractVTableFnTrait>);
+
+#[cfg(feature = "testutils")]
+impl Clone for ContractVTableFn {
+    fn clone(&self) -> Self {
+        self.0.duplicate()
+    }
+}
+
+#[cfg(feature = "testutils")]
+#[derive(Clone)]
+pub struct ContractVTable(pub std::collections::HashMap<Symbol, ContractVTableFn>);
```

Comment:
What is a contract in this context, other than a vtable?

Reply:
I don't think we know the answer to that yet.

If you prefer to keep the narrower focus I suggest `ContractFunctionSet`. I suggest that because if a developer asks me what the ContractVTable is, not knowing what a vtable is, I would tell them it is the functions of the contract.

I mostly want to remove the term vtable from the public API of the SDK. So even if it was only a vtable, I don't think we need to surface the term vtable. This is part of the API that contract developers will use. The term vtable is only common in certain technology ecosystems and it is not meaningful to probably a large variety of developers. I hope we can wrap this inside a macro, but even then I think the term vtable is unnecessary. It's simple to say that this is a `Contract`, or that this is a `ContractFunctionSet`.

Reply:
Done.